### ISSUE TITLE
Convert Metanorma XML to IEEE STL XML format

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/mnconvert#161]
Metanorma XML to IEEE STL XML format conversion.

IEEE has provided 3 sample XML documents to serve as conversion masters:
* https://github.com/metanorma/mn-samples-ieee-private

Comparison done by @Intelligent2013 here:
* https://github.com/metanorma/mn-samples-ieee-private/issues/3#issuecomment-1186114303

_Originally posted by @Intelligent2013 in https://github.com/metanorma/mn-samples-ieee-private/issues/3#issuecomment-1186114303_

**Comment by Intelligent2013** [comment]
Need update IEEE XML to Adoc before the MN XML to IEEE XML conversion:
- `:doctype:` should be extract from the title `IEEE Guide for the `, but not from `@content-type`;
- no adoc document's attribute for ICS description (example: `Character sets and information coding`), should be added into metanorma;
- add processing for `contrib-group/contrib/collab-alternatives`.

**Comment by Intelligent2013** [comment]
There is a discrepancy between IEEE DTD documentation and xml instance.

For the attribute `@content-type`:
<html><body>


guide | This document is an IEEE Guidelines document.
-- | --



</body>
</html>

but in master-1-standard.xml  - `content-type="standard"`:
```xml
<standards-document ... content-type="standard" ...
...
   <std-main-title>IEEE Guide for the 
```

**Comment by Intelligent2013** [comment]
In the example document there is warning section in the body of document:
```xml
<body>
		<warning>
			<p>IMPORTANT NOTICE: IEEE Standards documents are not intended to ensure safety, health, or
				environmental protection, or ensure against interference with or from other devices or
				networks. Implementers of IEEE Standards documents are responsible for determining and
				complying with all appropriate safety, security, environmental, health, and interference
				protection practices and all applicable laws and regulations.</p>
			<p>This IEEE document is made available for use subject to important notices and legal
				disclaimers. These notices and disclaimers appear in all publications containing this
				document and may be found under the heading &#x201c;Important Notice&#x201d; or
				&#x201c;Important Notices and Disclaimers Concerning IEEE Documents.&#x201d; They can also
				be obtained on request from IEEE or viewed at 				<uri>http://standards.ieee.org/IPR/disclaimers.html</uri>.</p>
		</warning>
		<sec id="sec1">
			<label>1.</label>
			<title>Overview</title>
```
PDF:
![image](https://user-images.githubusercontent.com/5559926/180035293-d8239e7f-289d-4873-93ac-e64a974112d3.png)

It contains the sentences from the boilerplate text (preface sections) of the latest available document (IEEE 2830-2021). I don't think that we should support 'warning' special section. @ronaldtse what is your opinion?

**Comment by ronaldtse** [comment]
@Intelligent2013 I believe this is no longer used in the new Style Manual. Let's not support this.

**Comment by Intelligent2013** [comment]
Update typo: **shouldn't** support

**Comment by Intelligent2013** [comment]
In the source IEEE master-1-standard.xml there is `<styled-content style="color" style-type="red">`
```xml
<sec id="sec1.stdstaff">
<title>Case-study 1: text in color</title>
<p>Normal text, <styled-content style="color" style-type="red">color text[use HTML convention of style:color-name]</styled-content>, normal text.</p>
</sec>
```

In the original PDF there aren't these Case-studies, and in the Style Manual there isn't information about color.

@ronaldtse should we support 'color' property?

**Comment by Intelligent2013** [comment]
In master-2-standards-distribution.xml there is a fragment with `editing-instruction`s relate to the whole list and the list item:
```xml
<std-def-list>
	<editing-instruction>
		<p>This instruction is for the whole list</p>
	</editing-instruction>
	<std-def-list-item>
		<editing-instruction>
			<p>This instruction is only for this item.</p>
		</editing-instruction>
		<term>ATE</term>
		<def>
			<p>automatic test equipment</p>
		</def>
	</std-def-list-item>
```

Currently, resulted adoc looks so:
```adoc
EDITOR: This instruction is for the whole list

EDITOR: This instruction is only for this item.
```

I don't figure out how to realize it.

**Comment by Intelligent2013** [comment]
In the source IEEE XMLs the word `Equation` placed sometimes  outside on `xref`:
```xml
<p>For far field effect, Equation <xref ref-type="disp-formula" rid="deqn1">(1)</xref> has been
```
sometimes inside:
```
in meters, <xref ref-type="disp-formula" rid="deqn1">Equation (1)</xref> remains unchanged
```

Therefore we get double `Equation` in the resulted MN xml -> IEEE xml:
```xml
<p>For far field effect, Equation <xref ref-type="disp-formula" rid="deqn1">Equation (1)</xref> has been
```
To do: remove `Equation` before `xref` in IEEE xml -> adoc conversion.

**Comment by Intelligent2013** [comment]
The LaTeX equation includes the number in IEEE xml as `\tag{1}`:
```xml
<disp-formula id="deqn1">
	<tex-math notation="LaTeX">\[ N = 4.4 + dBT - 20\log \left( \frac{D}{WH} \right)	\tag{1} \]</tex-math>
</disp-formula>
```
To do: in adoc, the equations without `\tag{...}` should be marked as 'unnumbered'.

**Comment by Intelligent2013** [comment]
Just for memory, there is a non-existing date in the source IEEE xml `master-2-standards-distribution.xml`:
```xml
<pub-date date-type="current-version" iso-8601-date="2013-02-29">
<day>29</day>
<month>February</month>
<year>2013</year>
</pub-date>
```
It causes the processing error (https://github.com/metanorma/metanorma-ieee/issues/126).

**Comment by Intelligent2013** [comment]
In `master-2-standards-distribution.xml` there is discrepancy between `contrib-group` item and and participants list.
`contrib141` relates to the group `IEEE-SA Standards Board Liaisons`:
```xml
				<contrib contrib-type="member" id="contrib140">
					<name-alternatives>
						<string-name specific-use="display">
							<given-names>Howard L.</given-names>
							<surname>Wolfman</surname>
						</string-name>
					</name-alternatives>
				</contrib>
			</contrib-group>
			<contrib-group content-type="IEEE-SA Standards Board Liaisons">
				<contrib contrib-type="nrc-representative" id="contrib141">
					<name-alternatives>
						<string-name specific-use="display">
							<given-names>Satish K.</given-names>
							<surname>Aggarwal</surname>
						</string-name>
					</name-alternatives>
					<role>NRC Representative</role>
				</contrib>
```

but in the participants list it relates to the previous group `IEEE-SA Standards Board approved`:
```xml
			<participants-sec>
				<p>When the IEEE-SA Standards Board approved this standard on 27 September 2007,
it had the following membership:</p>
					...
					<list-item>
						<p>
							<xref ref-type="contrib" rid="contrib141"/>
						</p>
					</list-item>
				</list>
			</participants-sec>
			<participants-sec>
				<p>Also included are the following nonvoting IEEE-SA Standards Board liaisons:</p>
				<officers>
					<list>
						<list-item>
							<p>
								<xref ref-type="contrib" rid="contrib142"/>
							</p>
						</list-item>
```

Currently, IEEE XML to MN ADOC conversion in mnconvert based on `participants-sec` lists, therefore in round-trip conversion to IEEE xml we get the difference between source and generated IEEE xml - `contrib141` "moved" to the previous group:
```xml
				<contrib contrib-type="nrc-representative" id="contrib141">
					<name-alternatives>
						<string-name specific-use="display">
							<given-names>Satish K.</given-names>
							<surname>Aggarwal</surname>
						</string-name>
					</name-alternatives>
					<role>NRC Representative</role>
				</contrib>
			</contrib-group>
			<contrib-group content-type="IEEE-SA Standards Board liaisons">
				<contrib contrib-type="nist-representative" id="contrib142">
					<name-alternatives>
						<string-name specific-use="display">
							<given-names>Michael H.</given-names>
							<surname>Kelley</surname>
						</string-name>
					</name-alternatives>
					<role>NIST Representative</role>
				</contrib>
```

**Comment by Intelligent2013** [comment]
Also, there are duplicated `contrib144` items:
```xml
<list-item>
		<p>
			<xref ref-type="contrib" rid="contrib144"/>
		</p>
	</list-item>
	<list-item>
		<p>
			<xref ref-type="contrib" rid="contrib144"/>
		</p>
	</list-item>
</list>
</officers>
</participants-sec>

```

**Comment by Intelligent2013** [comment]
There is difference in definitions numbering between master-1-standard.pdf  (without numbering):
![image](https://user-images.githubusercontent.com/5559926/183249731-934771cf-d479-4bf3-af0d-f04fa9d7b466.png)

and master-2-standard.pdf  (with numbering):
![image](https://user-images.githubusercontent.com/5559926/183249760-160c4ff1-7ab7-45fa-acd9-319aad8b42c4.png)

Currently, mnconvert doesn't put numbers into mn semantic xml->IEEE xml conversion.

**Comment by Intelligent2013** [comment]
> In the source IEEE master-1-standard.xml there is `<styled-content style="color" style-type="red">`
```xml
<sec id="sec1.stdstaff">
<title>Case-study 1: text in color</title>
<p>Normal text, <styled-content style="color" style-type="red">color text[use HTML convention of style:color-name]</styled-content>, normal text.</p>
</sec>
```
> In the original PDF there aren't these Case-studies, and in the Style Manual there isn't information about color.

Metanorma supports now a CSS style directive (https://github.com/metanorma/metanorma.org/commit/0a3a7802a31fd3464555c6bf7e39e3489f868a19), so `styled-content` will be encoded so:
```
[css color:red]#color text[use HTML convention of style:color-name]#
```

https://github.com/metanorma/metanorma-standoc/issues/647

**Comment by Intelligent2013** [comment]
From https://github.com/metanorma/metanorma-ieee/issues/116:

> add a 'special case' for preface section processing in MN XML -> IEEE XMl conversion:
> 
> * if  `@id` starts-with `front-ack` or
> * title `Dedication`
>   then convert it into the element `ack`.

**Comment by Intelligent2013** [comment]
In IEEE xml samples there are two formats for the references in the text body to the standard:
- in master-1-standard.xml - the prefix `Std` inside `pub-id`:
```xml
<mixed-citation>
  <std>
    <std-organization>IEEE</std-organization>
    <pub-id pub-id-type="std-designation">Std 979</pub-id>&#x2122;</std>
</mixed-citation>
```
- in master-2-standards-distribution.xml  - the prefix `Std` before `pub-id`:
```xml
<mixed-citation>
  <std>
    <std-organization>IEEE</std-organization> Std <pub-id pub-id-type="std-designation">1671</pub-id>
  </std>
</mixed-citation>
```

mnconvert will support only last one.

**Comment by Intelligent2013** [comment]
There is a discrepancy between original PDF and IEEE xml for 2830-2021 in part of Participants list.
In PDF there is list of 'Organization Represented' and 'Name of Representative':
![image](https://user-images.githubusercontent.com/5559926/184370849-cae52aed-4032-454a-93ff-214acc687240.png)

In XML there isn't information for 'Name of Representative' at all:
```xml
<contrib-group content-type="Working Group">
	<contrib id="contrib1" contrib-type="chair">
		<name-alternatives>
			<string-name>
				<given-names>Jin</given-names>
				<surname>Peng</surname>
			</string-name>
		</name-alternatives>
		<role>Chair</role>
	</contrib>
	<contrib id="contrib2" contrib-type="vice-chair">
		<name-alternatives>
			<string-name>
				<given-names>Cheng</given-names>
				<surname>Hong</surname>
			</string-name>
		</name-alternatives>
		<role>Vice Chair</role>
	</contrib>
	<contrib id="contrib3" contrib-type="member">
		<collab-alternatives>
			<collab>Alibaba China Co. Ltd.</collab>
		</collab-alternatives>
	</contrib>
	<contrib id="contrib4" contrib-type="member">
		<collab-alternatives>
			<collab>Alipay (China) Technology Co., Ltd</collab>
		</collab-alternatives>
	</contrib>
	<contrib id="contrib5" contrib-type="member">
		<collab-alternatives>
			<collab>Arpacorp Limited</collab>
		</collab-alternatives>
	</contrib>
...
```

```

**Comment by Intelligent2013** [comment]
There isn't one approach in the representation of `Acronyms and abbreviations`:
- in master-2-standards-distribution.xml - the element `std-def-list` is using:
```xml
<sec id="sec3.2">
	<label>3.2</label>
	<title>Acronyms and abbreviations</title>
	<std-def-list>
		<std-def-list-item>
			<term>ATE</term>
			<def>
				<p>automatic test equipment</p>
			</def>
		</std-def-list-item>
```

- in 2830-2021.xml - the element `def-list` is using:
```xml
<sec id="sec3.2">
	<label>3.2</label>
	<title>Acronyms and abbreviations</title>
	<def-list>
		<def-item>
			<term>API</term>
			<def>
				<p>application programming interface</p>
			</def>
		</def-item>
```

Currently, mnconvert uses 1st approach.

**Comment by Intelligent2013** [comment]
The conversion (except some issues) from IEEE XML to Metanorma adoc, and from Metanorma XML to IEEE xml available in mnconvert v1.20.0 (https://github.com/metanorma/mnconvert/releases/tag/v1.20.0).

How to convert from IEEE XML to Metanorma adoc:
```
java -jar mnconvert-1.20.0.jar master-1-standard.xml --output master-1-standard.adoc --imagesdir ./
```

How to convert from Metanorma XML (semantic) to IEEE XML:
```
java -jar mnconvert-1.20.0.jar document.xml --output-format ieee --validation-against C:\Metanorma\IEEE\standards-1-7-DTD\standards-1-7-dtd\standards.dtd
```

Latest documentation with mapping between IEEE XML and Metanorma adoc and XML: [IEEE_vs_NISO_and_MN_comparison_v.0.4.docx](https://github.com/metanorma/mnconvert/files/9331516/IEEE_vs_NISO_and_MN_comparison_v.0.4.docx)

The current ticket is too long, further remained issues will be solved in another tickets (see links above between posts).
